lib/foreman/event_subscribers.rb

```diff
@@ -0,0 +1,7 @@
+module Foreman
+  module EventSubscribers
+    def self.all_observable_events
+      ActiveRecord::Base.descendants.select { |klass| klass <= ::Foreman::EventSubscribers::Observable }.map(&:event_subscription_hooks).flatten.uniq
+    end
```

Comment:
This is unused yet, however we should totally memorize this method - it can be slow.

Reply:
It's used in the webhook plugin, but sure. We can memoize this.